### PR TITLE
WIP: Patch memory leaks in gwdetchar-omega

### LINKED ISSUE #93: Huge memory leaks in gwdetchar-omega
There are substantial memory leaks in the new `gwdetchar-omega` code:

```
top - 19:17:06 up 83 days, 8 min,  6 users,  load average: 1.41, 0.85, 0.74
Tasks: 1834 total,   3 running, 1826 sleeping,   5 stopped,   0 zombie
%Cpu(s):  6.4 us,  6.7 sy,  0.0 ni, 86.5 id,  0.3 wa,  0.0 hi,  0.1 si,  0.0 st
KiB Mem : 13175226+total, 64782468 free, 23777292 used, 43192504 buff/cache
KiB Swap: 16760828 total, 12701648 free,  4059180 used. 10513886+avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                     
2853460 aurban    20   0 6783288   4.5g  27552 R 280.0  3.6   1:02.19 gwdetchar-omega  
```

The code needs to be properly profiled, and these leaks need to be sealed up. For reference, both `dmt_omega` and the Matlab version never use much more than 2 GB per run.

### PR DESCRIPTION
**Pull request by alurban**
This PR is an attempt at reducing memory usage in `gwdetchar-omega` with the following changes:

* Read in only as much data as is requested by the user with the `duration` INI keyword
* Read in only one channel at a time so that no data get cached outside the loop in which they are used

Example output will be posted soon. This PR is a work in progress because it depends on #123 and #118. This fixes #93.

### CODE REVIEW

**Comment by alurban** [comment]
cc @duncanmmacleod @areeda

Last night I ran a test with this branch and tracked the memory usage:

Log | Max memory | Run time | Results
-- | -- | -- | --
[L1-pyomega-test](https://ldas-jobs.ligo-la.caltech.edu/~aurban/pyomega-test/log/L1-pyomega-test.log) | 20914 MB | 01:43:26 | [results](https://ldas-jobs.ligo-la.caltech.edu/~aurban/pyomega-test/L1_1172489783.07/)

Confusingly, this run used almost twice as much memory as the last test I did without these changes, but on the same IFO and GPS time (see the discussion in #93). The longer runtime was expected because of the change from front-loading data with `TimeSeriesDict.get()` to reading one channel at a time with `TimeSeries.get`.

Here is a summary of the big code differences between the two runs:

* Use the latest development branch of GWPy
* Move plot saving and closing into `gwdetchar.omega.plot`
* Read one channel at a time with `TimeSeries.get`, never caching data outside the loop in which they are analyzed
* Read in only the INI-configured `duration` seconds of data

The only thing I can think is _maybe_ I should be explicitly deleting plot objects at the end of `gwdetchar.omega.plot`,  which does not have a return statement. But (1) I am closing them with `plot.close()` and (2) I would have thought the garbage collector would take care of that. Any thoughts?

**Comment by duncanmmacleod** [comment]
One possible explanation is that there is a memory leak in the frame reader, since in this new scheme the TOC for the frame (the largest component) will be read once per channel, rather than once per block. Can you isolate that change and try another comparison?

File: gwdetchar/omega/plot.py [hunk 1]
```diff
@@ -66,18 +65,14 @@ def omega_plot(series, gps, span, channel, output, colormap='viridis',
     ax = plot.gca()
     # set time axis units
     ax.set_epoch(gps)
-    ax.set_xscale('auto-gps')
-    if span <= 1.:
-        ax.set_xlabel('Time [milliseconds]')
-    else:
-        ax.set_xlabel('Time [seconds]')
+    ax.set_xscale('auto-gps', epoch=gps)
+    ax.set_xlabel('Time [seconds]')
```
[the remark this answers is not in the document]
Comment (alurban) [reply gwdetchar/omega/plot.py]: @duncanmmacleod, I'll post an issue and do this in a separate PR in order to keep things focused.

**Comment by duncanmmacleod** [comment]
@alurban, what is the status of this PR?

**Pull request closed without merging**